File: Mall-UI/src/app/models/category.ts
import { MenuProduct, Product } from "./product"

export interface Category {
    id: number
    name: string
    image: string
    departmentId: number
    departmentName: string
}

export interface MenuCategory {
    id: number
    name: string
    products: MenuProduct[]
}

export interface CategoryCreationDTO {
    name: string
    image: File
    departmentId: number
}

File: Mall-API/Mall-API/Mall-API/Entities/Category.cs
    namespace Mall_API.Entities
{
    public class Category
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Image { get; set; }
        public int DepartmentId { get; set; }
        public Department Department { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}
